_posts/2022-08-29-sbt-tutorial.md

```diff
@@ -0,0 +1,496 @@
+
+Now that we added the dependency, we can add a test file as:
+```
+package com.rockthejvm
+import org.scalatest.funsuite.AnyFunSuite
+class SimpleTest extends AnyFunSuite {
+  test("comapare 2 strings ignoring case") {
+    val calculatedString = "ROCKtheJVM"
+    val expectedString = "rockthejvm"
+    assert(calculatedString.toLowerCase == expectedString)
+  }
+}
+```
+
+Now, we can run the tests in the project using the command `test`. This will run all the tests that are available in the project under the _src/test/scala_ path. 
+
+It is also possible to run a subset of tests instead of all. For example, let's try to run only one test file. For this we will first start the SBT session. Then within the session we can run:
+```
+test:testOnly com.rockthejvm.SimpleTest
+```
+This will test only the _SimpleTest_ file. Note that we need to provide the full package path to resolve the class correctly.
+We can also use a wildcard to make it easy to run the test:
+```
+test:testOnly *SimpleTest
+```
+This will run all the classes that ends with _SimpleTest_ within the entire project irrespective of the package.
+
+So far, we started the session and then ran the test. We can do that directly from the terminal by using the command:
+```
+sbt test 
+```
+This will start the SBT session and then run the test. However, we need to wrap the arguments after _sbt_ if there are some special characters are involved. For example, to run the test using wildcard, we need to use the command:
+```
+sbt 'test:testOnly *SimpleTest'
+```
+We can also use double quotes(") instead of single quotes('). This will ensure that all the arguments are passed to the SBT session at once.
+
+We can compile only the test classes by using the command `test:compile` within sbt session. 
+
+## 8. Advanced configuration in _build.sbt_
+In the previous section, we saw configurations like _name_, _organization_, etc. The values we set there are applicable for the entire project. SBT also let's us to do some configurations per scope. 
+For example, by default SBT runs all the tests in parallel. If we somehow want to avoid that and run the tests in sequential way, we can set the config option for that. SBT uses the configuration key _parallelExecution_ for that and it is under the scope of Test. 
+
+So, in _build.sbt_, we can use the setting as:
+```
+Test / parallelExecution := false
+```
+This will set the value to false within scopes of test.
+
+There are many such configurations which are configured using the scope. You can think of this something in the lines of _Test.parallelExecution_.
+
+SBT also has other different scopes such as `IntegrationTest`, `RunTime`, `Compile`, and so on. So, we can set the configuration only for a particular scope. For example, we can set some configurations only for the scope `IntegrationTest` as:
```

Comment:
Is there an exhaustive list of scopes? This leaves me (as a reader) wonder if there are any more. I like closure :D

Reply:
changed the sentence with all the scopes as list